### PR TITLE
feat: introduce `Action3P` for explicit 3-player action encoding

### PR DESCRIPTION
**Pull request by smly**
Introduce a dedicated `Action3P` wrapper type that always encodes actions using the 3-player action encoder, eliminating the need to manually pass `ActionEncoder::ThreePlayer` at every call site.

### Motivation

Previously, 3-player action encoding was handled by explicitly constructing an `ActionEncoder::ThreePlayer` at each usage site and calling `encoder.encode(action)`. This was error-prone: forgetting to use the correct encoder (or accidentally using the 4-player encoder) would silently produce incorrect action indices. By encapsulating the encoding strategy into the `Action3P` type itself, the correct encoder is always used and call sites become simpler (`action.encode()`).

### Summary

- Update `Observation3P` to store and return `Action3P` instead of `Action`, ensuring legal actions, masks, and action lookups consistently use 3-player encoding.
- Update `KyokuStepIterator3P` (replay iterator) to yield `Action3P` objects so downstream Python training code receives correctly-typed actions.
- Add an `AnyAction` adapter in the Python bindings so that `RiichiEnv.step()` accepts both `Action` and `Action3P`, preserving backward compatibility.
- Improve error handling in `MCDataset` to skip unparseable replays instead of crashing the training loop.
- Fix `q_values` extraction in `bc_logs.py` trainer to handle models that return a tuple.

### CODE REVIEW

File: riichienv-core/src/observation_3p/python.rs [hunk 1]
```diff
@@ -246,7 +245,7 @@ impl Observation3P {
 
         let actions_py = pyo3::types::PyList::empty(py);
         for a in &self._legal_actions {
-            actions_py.append(a.to_dict_py(py)?)?;
+            actions_py.append(a.0.to_dict_py(py)?)?;
```
[the remark this answers is not in the document]
Comment (smly) [reply riichienv-core/src/observation_3p/python.rs]: Fixed — now calling `a.to_dict_py(py)` directly, which resolves through Action3P's own `to_dict_py` method. This keeps the wrapper opaque as intended.

File: src/riichienv/_riichienv.pyi [hunk 2]
```diff
@@ -82,6 +82,16 @@ class Action:
     def to_dict(self) -> dict[str, Any]: ...
     def to_mjai(self) -> str: ...
 
+class Action3P:
+    action_type: ActionType
+    tile: int
+    consume_tiles: list[int]
+
+    def __init__(self, action_type: ActionType, tile: int = 0, consume_tiles: list[int] = []): ...
+    def to_dict(self) -> dict[str, Any]: ...
```
[the remark this answers is not in the document]
Comment (smly) [reply src/riichienv/_riichienv.pyi]: This mirrors the actual PyO3 constructor signature (`consume_tiles=vec![]`), which creates a fresh `Vec` on each call — there's no shared-state risk at runtime since the default is evaluated on the Rust side. The `.pyi` stub intentionally matches the runtime API for discoverability. That said, if this causes lint warnings in downstream projects, we can revisit.

File: riichienv-python/src/env.rs [hunk 3]
```diff
@@ -2,7 +2,26 @@ use pyo3::prelude::*;
...
+
+impl<'a, 'py> FromPyObject<'a, 'py> for AnyAction {
+    type Error = PyErr;
+
+    fn extract(obj: pyo3::Borrowed<'a, 'py, PyAny>) -> PyResult<Self> {
+        if let Ok(a) = obj.extract::<Action>() {
+            Ok(AnyAction(a))
+        } else if let Ok(a3p) = obj.extract::<Action3P>() {
+            Ok(AnyAction(a3p.0))
+        } else {
```
[the remark this answers is not in the document]
Comment (smly) [reply riichienv-python/src/env.rs]: Added a `From<Action3P> for Action` impl in `action.rs` and updated `AnyAction` to use `Action::from(a3p)` instead of `a3p.0`. This keeps the conversion idiomatic and decoupled from the internal representation.
